Set eviction-hard for vagrant cluster

dshulyak opened the pull request:
In order to test eviction related functionality it will be convenient to have reasonable eviction defaults.
At this moment exactly same flags are used by GCE environment

kubelet will have a following flag:

--eviction-hard=memory.available<100Mi,nodefs.available<10%

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32066)

<!-- Reviewable:end -->

dshulyak commented:
@derekwaynecarr do you think it is possible to land it this week? or better to wait?

derekwaynecarr commented on cluster/vagrant/config-default.sh:
```diff
@@ -115,3 +115,6 @@ E2E_STORAGE_TEST_ENVIRONMENT=${KUBE_E2E_STORAGE_TEST_ENVIRONMENT:-false}
 export DEFAULT_NETWORK_IF_NAME="eth0"
+
+# Evict pods whenever compute resource availability on the nodes gets below a threshold.
+EVICTION_HARD="${EVICTION_HARD:-memory.available<100Mi,nodefs.available<10%}"
```
nit: add a new line.

derekwaynecarr commented on cluster/vagrant/config-test.sh:
```diff
@@ -26,3 +26,6 @@ REGISTER_MASTER_KUBELET=${REGISTER_MASTER:-false}
 E2E_STORAGE_TEST_ENVIRONMENT=${KUBE_E2E_STORAGE_TEST_ENVIRONMENT:-false}
+
+# Evict pods whenever compute resource availability on the nodes gets below a threshold.
+EVICTION_HARD="${EVICTION_HARD:-memory.available<100Mi,nodefs.available<10%}"
```
nit: add new line

derekwaynecarr commented:
two nits, then this is good to go.  it does help testing 1.4 flakes, so i have no objection to getting this.

dshulyak committed:
Set eviction-hard for vagrant cluster

In order to test eviction related functionality it will be convenient
to have reasonable eviction defaults.
At this moment exactly same flags are used by GCE environment

kubelet will have a following flag:

  --eviction-hard=memory.available<100Mi,nodefs.available<10%

Change-Id: I56ca03bc3c5467c8450150e292f7a346fa7772a9
